scripts/with-go-mod.sh

```diff
@@ -6,28 +6,40 @@
 # when the command is finished. This script should be dropped when this
 # repository is a proper Go module with a permanent go.mod.
 
-set -e
+set -euo pipefail
 
 SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 ROOTDIR="$(cd "${SCRIPTDIR}/.." && pwd)"
 
-if test -e "${ROOTDIR}/go.mod"; then
-	{
-		scriptname=$(basename "$0")
-		cat >&2 <<- EOF
-			$scriptname: WARN: go.mod exists in the repository root!
-			$scriptname: WARN: Using your go.mod instead of our generated version -- this may misbehave!
-		EOF
-	} >&2
-else
+cleanup_paths=""
+create_symlink() {
+	local target="$1"
+	local link="$2"
+
+	pushd "${ROOTDIR}" > /dev/null
+	trap 'popd > /dev/null' RETURN
+
+	if [ -L "$link" ] && [ "$(readlink "$link")" = "$target" ]; then
+		# symlink already present; we're done
+		return
+	fi
+
+	if [ -e "$link" ]; then
+		echo "$(basename "$0"): WARN: $link exists but is not the expected symlink!" >&2
+		echo "$(basename "$0"): WARN: Using your version instead of our generated version -- this may misbehave!" >&2
+		return
+	fi
+
 	set -x
+	ln -s "$target" "$link"
+	cleanup_paths="$cleanup_paths ${ROOTDIR}/$link"
```

Comment:
```suggestion
	cleanup_paths+=( "${ROOTDIR}/$link" )
```

(frankly `"${ROOTDIR}/$link"` should go into a variable earlier in this function so it can just be used throughout)